client/fvESg_service.go

```diff
@@ -0,0 +1,389 @@
+	contList := models.ListFromContainer(cont,"fvRsProv")
+	st := &schema.Set{
+		F: schema.HashString,
+	}
+	for _, contItem := range contList{
+		dat := models.G(contItem, "tnVzBrCPName")
+		st.Add(dat)
+	}
+	return st, err
+}
+
+func (sm *ServiceManager) CreateRelationfvRsConsIfFromEndpointSecurityGroup( parentDn, tnVzCPIfName string) error {
+	dn := fmt.Sprintf("%s/rsconsIf-%s", parentDn, tnVzCPIfName)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s"				
+			}
+		}
+	}`, "fvRsConsIf", dn))
+	jsonPayload, err := container.ParseJSON(containerJSON)
+	if err != nil {
+		return err
+	}
+	req, err := sm.client.MakeRestRequest("POST", fmt.Sprintf("%s.json", sm.MOURL), jsonPayload, true)
+	if err != nil {
+		return err
+	}
+	cont, _, err := sm.client.Do(req)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%+v", cont)
+	return nil
+}
+
+func (sm *ServiceManager) DeleteRelationfvRsConsIfFromEndpointSecurityGroup(parentDn , tnVzCPIfName string) error{
+	dn := fmt.Sprintf("%s/rsconsIf-%s", parentDn, tnVzCPIfName)
+	return sm.DeleteByDn(dn , "fvRsConsIf")
+}
+
+func (sm *ServiceManager) ReadRelationfvRsConsIfFromEndpointSecurityGroup( parentDn string) (interface{},error) {
+	dnUrl := fmt.Sprintf("%s/%s/%s.json",models.BaseurlStr,parentDn,"fvRsConsIf")
+	cont, err := sm.GetViaURL(dnUrl)
+	contList := models.ListFromContainer(cont,"fvRsConsIf")
+	st := &schema.Set{
+		F: schema.HashString,
+	}
+	for _, contItem := range contList{
+		dat := models.G(contItem, "tnVzCPIfName")
+		st.Add(dat)
+	}
+	return st, err
+}
+
+func (sm *ServiceManager) CreateRelationfvRsCustQosPolFromEndpointSecurityGroup( parentDn, tnQosCustomPolName string) error {
+	dn := fmt.Sprintf("%s/rscustQosPol", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s","tnQosCustomPolName": "%s"
+								
+			}
+		}
+	}`, "fvRsCustQosPol", dn,tnQosCustomPolName))
+	jsonPayload, err := container.ParseJSON(containerJSON)
+	if err != nil {
+		return err
+	}
+	req, err := sm.client.MakeRestRequest("POST", fmt.Sprintf("%s.json", sm.MOURL), jsonPayload, true)
+	if err != nil {
+		return err
+	}
+	cont, _, err := sm.client.Do(req)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%+v", cont)
+	return nil
+}
+
+func (sm *ServiceManager) ReadRelationfvRsCustQosPolFromEndpointSecurityGroup( parentDn string) (interface{},error) {
+	dnUrl := fmt.Sprintf("%s/%s/%s.json",models.BaseurlStr,parentDn,"fvRsCustQosPol")
+	cont, err := sm.GetViaURL(dnUrl)
+	contList := models.ListFromContainer(cont,"fvRsCustQosPol")
+	if len(contList) > 0 {
+		dat := models.G(contList[0], "tnQosCustomPolName")
+		return dat, err
+	} else {
+		return nil,err
+	}
+}
+
+func (sm *ServiceManager) CreateRelationfvRsConsFromEndpointSecurityGroup( parentDn, tnVzBrCPName string) error {
+	dn := fmt.Sprintf("%s/rscons-%s", parentDn, tnVzBrCPName)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s"				
```

Comment:
We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.